src/box/txn.c

```diff
@@ -653,6 +653,8 @@ txn_commit_stmt(struct txn *txn, struct request *request)
 		assert(stmt->row != NULL);
 		txn_update_row_counts(txn, stmt, 1);
 	}
+	if (stmt->space != NULL && space_is_data_temporary(stmt->space))
```

Comment:
Isn't it complement of the condition above? It could be just `else` clause.

Reply:
No, the condition above has an `or`. We shouldn't set the flag if the statement doesn't have a space assigned to it.

Reply:
True, thanks!